libcontainer/cgroups/systemd/common.go

```diff
@@ -351,21 +356,61 @@ func stopUnit(dbusConnection *systemdDbus.Conn, unitName string) error {
 	return nil
 }
 
+func systemdVersion() (int, error) {
```

Comment:
This seems heavy to call per runc invocation. Should we think about using build tags instead, so users can compile it for their version of systemd as needed? Another option may be a command-line flag. This applies to any other systemd feature detection as well.

Reply:
Let's just always assume systemd >= 242, and fall back to 241-compatible mode on error

Reply:
> This seems heavy to call per runc invocation

Does not look like heavy to me, and it's not per runc invocation, but only upon container creation or update (and in this case we talk to dbus already to create/update a unit, so there's no such thing as creating and new connection just to get the version).

Or am I missing something here?

The alternative is to exec systemctl and parse its output. I haven't checked which way is faster, but can look into it.

> Let's just always assume systemd >= 242, and fall back to 241-compatible mode on error

I was thinking about it, but the logic would look really ugly (got EPERM? remove cpu quota period and retry). I would consider it if there is no other way to know if systemd supports the feature or not, but it seems